src/PayloadValidator.cs

```diff
@@ -0,0 +1,121 @@
+            {
+                string fieldName = config.FieldName;
+                foreach (var validationType in config.Validations)
+                {
+                    ApplyValidation(request, fieldName, validationType);
+                }
+            }
+        }
+
+        private void ApplyValidation(Dictionary<string, object> payload, string field, ValidationType validationType)
+        {
+            switch (validationType)
+            {
+                case ValidationType.NON_NULL:
+                    ValidateNonNull(payload, field);
+                    break;
+                case ValidationType.NON_EMPTY_STRING:
+                    ValidateNonEmptyString(payload, field);
+                    break;
+                case ValidationType.URL:
+                    ValidateUrl(payload, field);
+                    break;
+                case ValidationType.ID:
+                    ValidateId(payload, field);
+                    break;
+                case ValidationType.MODE:
+                    ValidateMode(payload, field);
+                    break;
+                case ValidationType.TOKEN_GRANT:
+                    ValidateTokenType(payload, field);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private void ValidateMode(Dictionary<string, object> payload, string field)
+        {
+            if (!payload.ContainsKey(field))
+            {
+                return;
+            }
+            string mode = (string)payload[field];
+            if (!new List<string> { "test", "live" }.Contains(mode))
+            {
+                string errorMessage = "Invalid value provided for field {0}";
+                throw new BadRequestError(string.Format(errorMessage, field), "BAD_REQUEST_ERROR", 400);
+            }
+        }
+
+        private void ValidateTokenType(Dictionary<string, object> payload, string field)
+        {
+            ValidateNonNull(payload, field);
+            string grantType = (string)payload[field];
+            switch (grantType)
+            {
+                case "authorization_code":
+                    ValidateNonNull(payload, "code");
+                    ValidateNonNull(payload, "redirect_uri");
+                    break;
+                case "refresh_token":
+                    ValidateNonNull(payload, "refresh_token");
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private void ValidateNonNull(Dictionary<string, object> payload, string field)
+        {
+            if (!payload.ContainsKey(field) || payload[field] == null)
+            {
+                string errorMessage = "Field {0} cannot be null";
+                throw new BadRequestError(string.Format(errorMessage, field), "BAD_REQUEST_ERROR", 400);
+            }
+        }
+
+        private void ValidateNonEmptyString(Dictionary<string, object> payload, string field)
```

Comment:
ValidateNonEmptyString and ValidateNonNull can be clubbed

Reply:
IMO no. For some cases, we are checking that a field should exist. In some cases, field can be optional, but if passed, value cant be empty string.